src/modules/launcher/PowerLauncher/MainWindow.xaml.cs

```diff
@@ -126,7 +126,16 @@ public IntPtr ProcessWindowMessages(IntPtr hwnd, int msg, IntPtr wparam, IntPtr
         private void OnSourceInitialized(object sender, EventArgs e)
         {
             // Initialize protected environment variables before register the WindowMessage
-            EnvironmentHelper.GetProtectedEnvironmentVariables();
+            try
+            {
+                EnvironmentHelper.GetProtectedEnvironmentVariables();
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                Log.Exception("ErrorReporting", ex, typeof(EnvironmentHelper));
+            }
```

Comment:
@jaimecbernardo 
I have added this for testing. Without that the process crashes silently when we have an exception in `EnvironmentHelper.GetProtectedEnvironmentVariables()`. How should we handle this finally?

Possible solution:
- Leave as is now.
- Remove `try-catch`and ignore the possibility of silent crashes.
- Adding multiple try-catch blogs in `EnvironmentHelper.GetProtectedEnvironmentVariables()`.

Reply:
I think it's best to add try-catch inside the `EnvironmentHelper.GetProtectedEnvironmentVariables()`, logging the exception as you are doing (with a more descriptive string).
Have you got some stack traces from what's happening yet?

Reply:
The exceptions I got were bug that I made (type-cast). They are fixed.
And some casing problems with variable names when writing dictionary entry (Path <> PATH) which I will investigate and fix.

But it might make sense to move `System.Environmemt.GetEnvironmentVariables` into a `try-catch` block and stop if we can't get the variables. (I don't think that we might have other crashes, if type-cast and case-sensitive problems are fixed.)